src/com/goide/highlighting/GoAnnotator.java

```diff
@@ -110,6 +110,47 @@ else if (element instanceof GoBuiltinCallExpr) {
         checkMakeCall(call, holder);
       }
     }
+    else if (element instanceof GoFunctionLit) {
+      checkLiteralEvaluatedButNotUsed(holder, element, "Function");
+    }
+    else if (element instanceof GoStringLiteral) {
+      checkLiteralEvaluatedButNotUsed(holder, element, "String");
+    }
+    else if (element instanceof GoLiteral) {
+      String message = ((GoLiteral)element).getChar() != null ? "Rune" : "Number";
+      checkLiteralEvaluatedButNotUsed(holder, element, message);
+    }
+  }
+
+  private static void checkLiteralEvaluatedButNotUsed(@NotNull AnnotationHolder holder, @NotNull PsiElement element, String message) {
+    PsiElement parent = element.getParent();
+    if (element instanceof GoFunctionLit) {
+      if (parent instanceof GoCallExpr) return;
+    }
+    else if (element instanceof GoStringLiteral ||
+             element instanceof GoLiteral) {
+      if (parent instanceof GoIndexOrSliceExpr ||
+          parent instanceof GoConditionalExpr) {
+        return;
+      }
+    }
+
+    parent = PsiTreeUtil.getParentOfType(element, GoLeftHandExprList.class, GoArgumentList.class);
+    if (parent == null) return;
+    if (parent instanceof GoArgumentList) return;
+    if (parent instanceof GoLeftHandExprList &&
+        ((GoLeftHandExprList)parent).getExpressionList().size() != 1) {
+      return;
+    }
+
+    if (element instanceof GoFunctionLit) {
+      element = ((GoFunctionLit)element).getFunc();
+    }
+    else if (parent.getParent().getParent() instanceof GoSwitchStatement) {
```

Comment:
Please run annotator on your code base and check that we haven't any false positive cases. 
Use Ctrl+Alt+Shift+I: annotator 
![screenshot 2015-12-24 12 19 39](https://cloud.githubusercontent.com/assets/426890/11992559/a683f5ca-aa38-11e5-928d-dab276212c00.png)